plugins/libplugin-pay.c

```diff
@@ -1424,8 +1424,6 @@ static struct command_result *payment_createonion_success(struct command *cmd,
 	json_add_hex_talarr(req->js, "onion", p->createonion_response->onion);
 
 	json_object_start(req->js, "first_hop");
-	json_add_short_channel_id(req->js, "channel", &first->channel_id);
-	json_add_num(req->js, "direction", first->direction);
```

Comment:
This needs to be made dependent on `if (!deprecated_apis)` with a Changelog-Deprecated: JSON-RPC: `createonion` `first_hop` `channel` and `direction` fields removed.

Then we can actually rip it out in 6 months.

Reply:
But they are being ignored!

Or do you mean we can't remove because then the calls will fail for everybody that is currently passing these value?

Reply:
That's exactly what the `deprecated_apis` flag is for. If enabled it must replicate the old behavior, if disabled it should report the error.

Reply:
Sorry, but let me insist on this point I'm not understanding: this PR doesn't change the current behavior at all. It just stops passing two parameters (channel and direction) that were not being used at all. The behavior should be the exact same as before.

In this case, it doesn't make sense to me to guard the changes with a deprecation notice. What am I getting wrong about this?

Reply:
Let me point you towards the sea of red that the CI tests currently are. This PR breaks a lot of tests (if done in isolation). Please get the tests working again. The failures will point you towards the issue described by Rusty, since `sendonion` still requires the fields you removed.

Reply:
Huh, I misread.  This patch is fine: it just takes them out of the internal call that pay uses.

Of course, CI says that breaks everything, so we should fix that! 

```patch
diff --git a/lightningd/pay.c b/lightningd/pay.c
index d96686beb..7c34b4d87 100644
--- a/lightningd/pay.c
+++ b/lightningd/pay.c
@@ -1026,7 +1026,7 @@ send_payment_core(struct lightningd *ld,
 					   "peer found");
 
 		json_add_routefail_info(data, 0, WIRE_UNKNOWN_NEXT_PEER,
-					&ld->id, channel->scid,
+					&ld->id, NULL,
 					node_id_idx(&ld->id, &first_hop->nodeid),
 					NULL);
 		json_object_end(data);
```

Reply:
Why this change? Shouldn't we pass the actual channel we're using here (`channel->scid`)?

Reply:
channel is NULL here, it's inside an "if (!channel)".

Reply:
Oops, thank you.